### ISSUE TITLE
ci: Nix enhancements

### ISSUE THREAD

**Issue by realeinherjar** [bitcoindevkit/bdk#1185]
Some enhancements were suggested in #1165.
They are non-blockers and I am moving them to a separate issue for the near future.

- [ ] Move to [`rustshop/flakebox`](https://github.com/rustshop/flakebox)
   It can build derivations in parallel with `pkgs.linkFarmFromDrvs` ([example](https://github.com/rustshop/flakebox/blob/5dea81cad2be257a2cbccbd42b2f1c0fdf69008d/checks/workspace-cross-compile/default.nix#L93), from https://github.com/bitcoindevkit/bdk/pull/1165#discussion_r1374071091)

**Comment by notmandatory** [comment]
I put these in the beta.0 release for now, but may move them to future milestone if not urgent / ready to work on now.
